protocols/modbus/src/main/resources/protocols/modbus/modbus.mspec

```diff
@@ -416,6 +416,15 @@
         ['WCHAR' List
             [array uint 16 value count 'numberOfValues']
         ]
+        ['STRING','1' STRING
+            [simple vstring '8' value encoding='"UTF-8"']
+        ]
+        ['STRING' List
+            [simple vstring '8 * numberOfValues' value encoding='"UTF-8"']
```

Comment:
Here you are creating only one single element field of type string that you want to put in a list ... I think you should instead split this up into two cases: numberOfValues = 1, where the type is STRING and one where it's not == 1 where the type is List and the value field then needs to be an array.

Reply:
It doesn't make much sense, cause output should be a `String` anyway. What's the point of returning a text in chunks to the caller if it can be merged and read at once beforehand?

Reply:
I think you are mis-understanding the concept here ... 
A field of type "string" or "vstring" can contain a string of multiple characters. But with "STRING[10]" we are not reading one String with a length of 10 characters, but 10 strings of individual length (In S7 this would be 255 chars long) ... if you want to explicitly control the string lenght that you want to read use round bracets ... so STRING(10)[20] would read 20 strings of each 10 chars length. You are currently mixing up the array size and the lenght.

Reply:
Ok, it makes more sense then, will check out how rounded brackets go and update/revert template change if needed.

Reply:
Have a look at the S7 protocol ... there we're already using the string-length.

Reply:
Looking at S7 fields I see that its a special case regexp which is not going to fly with modbus unless we modify its definition. Will try cooking something there.